tools/test-runner/src/buck_test.rs

```diff
@@ -0,0 +1,252 @@
+use std::collections::{HashMap, HashSet};
+use std::env;
+use std::fs::File;
+use std::io::{BufReader, Read};
+use std::path::PathBuf;
+use std::process::{Command, Stdio};
+use std::time::{Duration, Instant};
+
+use anyhow::{bail, Context, Result};
+use serde::Deserialize;
+
+use super::pyunit;
+use super::rust;
+
+/// Unit test.
+#[derive(Debug)]
+pub struct Test {
+    /// The command used to execute the test.
+    pub command: Command,
+
+    /// User-facing identifier for this specific test.
+    pub name: String,
+
+    /// Labels/tags associated to this test.
+    pub labels: HashSet<String>,
+
+    /// Contacts for further information.
+    pub contacts: HashSet<String>,
+
+    /// Which type of test this is.
+    pub kind: TestKind,
+}
+
+/// Supported types of tests.
+#[derive(Clone, Copy, Debug, Deserialize)]
+#[serde(rename_all = "lowercase")]
+pub enum TestKind {
+    /// Buck's custom `unittest` runner.
+    Pyunit,
+
+    /// Buck's custom Rust tester.
+    Rust,
+
+    /// Simply runs a command and checks for non-zero exit code.
```

Comment:
Did you end up with this from the tests in `antlir/vm/tests`?
Just curious since those should be being passed to the external test runner as the underlying type (`pyunit` or `rust` - or `gtest` for cpp but we said you can safely ignore that in this PR)

Reply:
Yup, these (`pyunit` and `rust`) are the type tags buck test passes in to the runner.
`shell` was something I added as a sort of default, since a plain `sh_test` won't include any type tags.

Reply:
Actually, a quick test with `sh_test` shows that buck will add "custom" as a tag if no other type is provided